src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs

```diff
@@ -0,0 +1,199 @@
+            Mandatory = true,
+            HelpMessage = ResourceGroupParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = true, 
+            HelpMessage = NameParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = AgentCountParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = AgentCountParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public int? AgentCount { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslStatusParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslStatusParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string SslStatus { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslCertificateParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslCertificateParameterHelpMessage)]
+        public string SslCertificate { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslKeyParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslKeyParameterHelpMessage)]
+        public string SslKey { get; set; }
+
+        [Parameter(ParameterSetName = CmdletParametersParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslCNameParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslCNameParameterHelpMessage)]
+        public string SslCName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, @"Updating operationalization cluster..."))
```

Comment:
@shutchings I would move the two if-statements that set `Name` and `ResourceGroupName` for the `ObjectParameterSet` and `ResourceIdParameterSet` sets above this line so that `Name` has an appropriate value before being printed by the `ShouldProcess` block

Reply:
Good catch.